### PR TITLE
[DOC-3070] OAuth with GitX

### PR DESCRIPTION
**Pull request by rashminandasahoo**
Topic on OAuth integration with GitX. **Preview**: https://646c4d30c49a3b241e7f0a48--harness-developer.netlify.app/
## What Type of PR is This?

- [ ] Issue
- [x ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ x] *Optional* Screen Shoot. 
connect-git-provider

### CODE REVIEW

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+
+:::important
+Currently, this feature is behind the feature flag `PIE_GITX_OAUTH`. Contact Harness Support to enable the feature.
+:::
+
+You can store configurations for your resources, such as pipelines and input sets, in Git using Harness Git Experience. In order to commit configuration changes to Git, Harness fetches the user credentials from the Git connector used to push the changes and uses them as the author.
+
+You can now push configuration changes using your own credentials by integrating OAuth with Git. Harness fetches your credentials from your user profile for all Git commits.
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/oauth-integration.md:14]: This line is bit vague: "Harness fetches your credentials from your user profile for all Git commits.". I think we can mention that whenever OAuth is integrated for a user, the credentials are stored as Harness User Profile. For any commits, these creds are used thereafter.

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+:::important
+Harness will continue to fetch Git connector credentials if you don't integrate OAuth with Git experience.
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/oauth-integration.md:17]: continue to "use"?

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+description: Configure your credentials for Git experience.
+sidebar_position: 10
+---
+
+
+:::important
+Currently, this feature is behind the feature flag `PIE_GITX_OAUTH`. Contact Harness Support to enable the feature.
+:::
+
+You can store configurations for your resources, such as pipelines and input sets, in Git using Harness Git Experience. In order to commit configuration changes to Git, Harness fetches the user credentials from the Git connector used to push the changes and uses them as the author.
```
[the remark this answers is not in the document]
Comment (shalini-agr) [reply docs/platform/10_Git-Experience/oauth-integration.md:12]: Harness account "user name" is used as the author for now.

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+This topic explains how to configure OAuth for Git experience in Harness.
+
+## Configure OAuth for Git provider
+
+Harness supports OAuth integration for the following Git providers: 
+* GitHub
+* GitLab
+* Bitbucket
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/oauth-integration.md:28]: Lets make it Bitbucket SAAS

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+   If you have not configured OAuth for Git, Harness prompts you to set up an access token in the user profile to use it for Git commits.
+   You can do one of the following to commit your changes to Git: 
+   * Select **Connect** to configure OAuth for Git and use your Git credentials for the commit and then select **Save**. This is optional.
+   * Select **Save** to use the credentials from the Git connector for the commit.
+
+   ![](./static/commit-without-oauth.png)
+   
+   :::
+
+4. Go to your branch in the Git repository. It now displays your credentials besides the commit.
```
Comment (mohitg0795) [inline docs/platform/10_Git-Experience/oauth-integration.md:71]: now displays your "author details" besides the commit.

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+   ![](./static/oauth-credentials.png)
+
+## Commit changes to Git with the configured OAuth token
+
+To commit changes to Git using the configured OAuth token: 
+
+1. Go to an existing remote pipeline or [create](./configure-git-experience-for-harness-entities.md#add-a-remote-pipeline) a new one.
+2. Edit the pipeline and select **Save**.
+   The **Save Pipelines to Git** settings appear. Harness displays the credentials being used for this commit.
```
[the remark this answers is not in the document]
Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/oauth-integration.md:55]: I prefer not to display such info in any of our docs.

[the remark this answers is not in the document]
Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/oauth-integration.md:55]: @mohitg0795  Shouldn't it be user name corresponding to the credentials?

File: docs/platform/10_Git-Experience/oauth-integration.md (hunk 5 above)
Comment (rashminandasahoo) [reply docs/platform/10_Git-Experience/oauth-integration.md]: Done

**Review by mohitg0795 — DISMISSED** [review]

**Review by shalini-agr — DISMISSED** [review]

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+---
+title: Integrate OAuth with Git experience
+description: Configure your credentials for Git experience.
+sidebar_position: 10
+---
+
+
+:::important
+Currently, this feature is behind the feature flag `PIE_GITX_OAUTH`. Contact Harness Support to enable the feature.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/oauth-integration.md:9]:
```suggestion
Currently, this feature is behind the feature flag `PIE_GITX_OAUTH`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
```

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+:::
+
+You can store configurations for your resources, such as pipelines and input sets, in Git using Harness Git Experience. Harness fetches user credentials from the Harness account and uses the corresponding user name as the author of the commit.
+
+You can now push configuration changes using your own credentials by integrating OAuth with Git.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/oauth-integration.md:14]:
```suggestion
You can push configuration changes using your own credentials by integrating OAuth with Git. 
```

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+
+:::important
+Currently, this feature is behind the feature flag `PIE_GITX_OAUTH`. Contact Harness Support to enable the feature.
+:::
+
+You can store configurations for your resources, such as pipelines and input sets, in Git using Harness Git Experience. Harness fetches user credentials from the Harness account and uses the corresponding user name as the author of the commit.
+
+You can now push configuration changes using your own credentials by integrating OAuth with Git. 
+When you integrate OAuth with Git experience, the credentials are stored in your Harness account user profile. These credentials are used for any subsequent commits.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/oauth-integration.md:15]:
```suggestion
When you integrate OAuth with Git Experience, the credentials are stored in your Harness account user profile. These credentials are used for any subsequent commits.
```

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+:::important
+Harness will continue to use the user name corresponding to the account if you don't integrate OAuth with Git experience.
+:::
+
+
+This topic explains how to configure OAuth for Git experience in Harness.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/oauth-integration.md:22]:
```suggestion
This topic explains how to configure OAuth for Git Experience in Harness.
```

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+1. Go to an existing remote pipeline or [create](./configure-git-experience-for-harness-entities.md#add-a-remote-pipeline) a new one.
+2. Edit the pipeline and select **Save**.
+   The **Save Pipelines to Git** settings appear. Harness displays the user name being used for this commit.
+
+   ![](./static/git-commit-oauth.png)
+
+3. Select **Save**.
+
+   :::important
+   If you have not configured OAuth for Git, Harness prompts you to set up an access token in the user profile to use it for Git commits.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/oauth-integration.md:63]:
```suggestion
   If you have not configured OAuth for Git, Harness prompts you to set up an access token in the user profile to use for Git commits.
```

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+   You can do one of the following to commit your changes to Git: 
+   * Select **Connect** to configure OAuth for Git and use your Git credentials for the commit and then select **Save**. This is optional.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/oauth-integration.md:65]:
```suggestion
   * Select **Connect** to configure OAuth for Git and use your Git credentials for the commit, and then select **Save**. This is optional.
```

File: docs/platform/10_Git-Experience/oauth-integration.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+   * Select **Save** to use the credentials from the Git connector for the commit.
+
+   ![](./static/commit-without-oauth.png)
+   
+   :::
+
+4. Go to your branch in the Git repository. It now displays the author details besides the commit.
```
Comment (brian-f-harness) [inline docs/platform/10_Git-Experience/oauth-integration.md:72]: Suggest either:
It now displays the author details beside the commit.
or 
It now displays the author details next to the commit.
or 
It now displays the author details with the commit.

**Review by brian-f-harness — DISMISSED** [review]
lgtm. Added a couple small suggestions.

**Pull request merged**